alchemy/src/neon/branch.ts

```diff
@@ -0,0 +1,673 @@
+import type { Context } from "../context.ts";
+import { Resource } from "../resource.ts";
+import { handleApiError } from "./api-error.ts";
+import { createNeonApi, type NeonApiOptions, type NeonApi } from "./api.ts";
+import type { NeonProject } from "./project.ts";
+
+/**
+ * Properties for creating or updating a Neon branch
+ */
+export interface NeonBranchProps extends NeonApiOptions {
+  /**
+   * The project containing the branch
+   */
+  project: string | NeonProject;
+
+  /**
+   * Name of the branch
+   */
+  name: string;
+
+  /**
+   * ID of the parent branch to create this branch from
+   */
+  parentId?: string;
```

Comment:
This should be `parent?: string | Branch`. When a resource's props refers to an domain-specific entity like this, it should always be done in a way that is "internal" to the alchemy abstraction and "external" (via `string` reference). Make sure to apply this same logic across all the new resources you just built.